src/library_noderawfs.js

```diff
@@ -96,16 +96,32 @@ mergeInto(LibraryManager.library, {
       var newMode = NODEFS.getMode(pathTruncated);
       var fd = suggestFD != null ? suggestFD : FS.nextfd(nfd);
       var node = { id: st.ino, mode: newMode, node_ops: NODERAWFS, path: path }
-      var stream = { fd: fd, nfd: nfd, position: 0, path: path, flags: flags, node: node, seekable: true };
+      var stream = FS.createStream({ fd: fd, nfd: nfd, position: 0, path: path, flags: flags, node: node, seekable: true });
       FS.streams[fd] = stream;
       return stream;
     },
+    createStream: function(stream, fd_start, fd_end){
+      // Call the original FS.createStream
+      var rtn = VFS.createStream(stream, fd_start, fd_end);
+      if (typeof rtn.shared.refcnt == "undefined") {
+        rtn.shared.refcnf = 0;
+      } else {
+        rtn.shared.refcnf++;
+      }
+      return rtn;
+   },
+   closeStream: function(fd) {
+    if(FS.streams[fd]){
```

Comment:
It seems like both the call sites to closeStream get the fd from a live stream object so its hard to see how this could ever fail?

```
src/library_fs.js:        FS.closeStream(stream.fd);
src/library_noderawfs.js:      FS.closeStream(stream.fd);
```

Perhaps could simplify the code by changed `closeStream` such that it takes the actal stream object and not the fd?

Perhaps that can be done as a followup since that PR has already been though many iterations.

Reply:
Well it fixed the failing `test_freetype`...